### PR TITLE
[openstack_horizon] Fix collection of horizon logs from /var/log/httpd

### PR DESCRIPTION
**Pull request by shatadru**
Fixes : #1131

with all_logs, all logs for respective plugins are collected
from  /var/log/httpd.

without all_logs only *.log files are collected.

Signed-off-by: Shatadru Bandyopadhyay <shatadru1@gmail.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

File: sos/plugins/openstack_aodh.py [hunk 1]
```diff
@@ -38,11 +38,17 @@ def setup(self):
 
         self.limit = self.get_option("log_size")
         if self.get_option("all_logs"):
-            self.add_copy_spec("/var/log/aodh/",
-                               sizelimit=self.limit)
+            self.add_copy_spec([
+                "/var/log/aodh/",
+                "/var/log/httpd/aodh*"],
+                sizelimit=self.limit
+            )
```
Comment (bmr-cymru) [inline sos/plugins/openstack_aodh.py]: Please fix this to place the closing `)` on the same line as the final keyword argument. This is the accepted style in `sos`.

Comment (bmr-cymru) [reply sos/plugins/openstack_aodh.py]: This should be written as:

```python
    self.add_copy_spec([
        "/var/log/aodh/",
        "/var/log/httpd/aodh*"
    ], sizelimit=self.limit)
```

File: sos/plugins/openstack_aodh.py [hunk 2]
```diff
@@ -38,11 +38,17 @@ def setup(self):
...
+
         else:
-            self.add_copy_spec("/var/log/aodh/*.log",
-                               sizelimit=self.limit)
+            self.add_copy_spec([
+                "/var/log/aodh/*.log",
+                "/var/log/httpd/aodh*.log"],
+                sizelimit=self.limit)
```
Comment (bmr-cymru) [inline sos/plugins/openstack_aodh.py]: Ditto.

File: sos/plugins/openstack_ceilometer.py [hunk 3]
```diff
@@ -92,4 +92,15 @@ class RedHatCeilometer(OpenStackCeilometer, RedHatPlugin):
         'python-ceilometerclient'
     )
 
+    def setup(self):
+        super(RedHatCeilometer, self).setup()
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/httpd/ceilometer*",
+            ], sizelimit=self.limit)
```
Comment (bmr-cymru) [inline sos/plugins/openstack_ceilometer.py]: This is the correct style to use in `sos`.

File: sos/plugins/openstack_ceilometer.py [hunk 4]
```diff
@@ -92,4 +92,15 @@ class RedHatCeilometer(OpenStackCeilometer, RedHatPlugin):
...
+        else:
+            self.add_copy_spec([
+                "/var/log/httpd/ceilometer*.log",
+            ], sizelimit=self.limit)
```
Comment (bmr-cymru) [inline sos/plugins/openstack_ceilometer.py]: Correct.

File: sos/plugins/openstack_horizon.py [hunk 5]
```diff
@@ -117,7 +117,14 @@ class RedHatHorizon(OpenStackHorizon, RedHatPlugin):
...
         self.add_copy_spec("/etc/httpd/conf.d/openstack-dashboard.conf")
-        if self.get_option("log"):
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/httpd/horizon*",
+            ], sizelimit=self.limit)
+        else:
+            self.add_copy_spec([
+                "/var/log/httpd/horizon*.log"
+            ], sizelimit=self.limit)
```
Comment (bmr-cymru) [inline sos/plugins/openstack_horizon.py]: Correct x 2.

File: sos/plugins/openstack_keystone.py [hunk 6]
```diff
@@ -120,4 +120,16 @@ class RedHatKeystone(OpenStackKeystone, RedHatPlugin):
...
+        super(RedHatKeystone, self).setup()
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/httpd/keystone*",
+            ], sizelimit=self.limit)
+        else:
+            self.add_copy_spec([
+                "/var/log/httpd/keystone*.log",
+            ], sizelimit=self.limit)
+
```
Comment (bmr-cymru) [inline sos/plugins/openstack_keystone.py]: Correct x 2.

**Comment by shatadru** [comment]
Hi @bmr-cymru,  amended the commit with suggested changes, let me know if it is fine.

**Comment by shatadru** [comment]
Hi @bmr-cymru : exactly! intention was to hard wrap the just the commit message and not the subject itself:( 

I have corrected it.

Thanks

**Comment by bmr-cymru** [comment]
Thanks! 👍

**Pull request closed without merging**